### PR TITLE
E402: Add "with" statement to allowed keywords

### LINKED ISSUE #833: E402: Add "with" to `allowed_try_keywords`
For certain imports, it would be nice to scope warning filters (e.g. in https://github.com/RobotLocomotion/drake/issues/10527).

In my case, I want to do something like:
```py
import warnings

with warnings.catch_warnings():
    warnings.filterwarnings(
        "ignore", category=ImportWarning,
        message="can't resolve package from __spec__")
    import meshcat
```

Without mods to 2.3.1, it throws a warning about this. If I add "with" to the allowed keywords, then it works.

Would it be possible to add this keyword?

### PR DESCRIPTION
**Pull request by EricCousineau-TRI**
Resolves #833

### CODE REVIEW

**Comment by asottile** [comment]
Should `if` be added as well?

**Comment by EricCousineau-TRI** [comment]
SGTM. Will file as a separate commit, and split it into a separate PR if it causes issues.

**Comment by EricCousineau-TRI** [comment]
Done.

**Comment by EricCousineau-TRI** [comment]
Ah, sorry about that! And thanks!

File: testsuite/E40.py [hunk 1]
```diff
@@ -32,6 +32,13 @@
...
     print('made attempt to import foo')
+#: Okay
+if False:
+    import foo
+#: Okay
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", DeprecationWarning)
+    import foo
 
 import bar
```
Comment (asottile) [inline testsuite/E40.py]: oh actually in order to demonstrate this test you need to have an `import bar` after each of the blocks that would have triggered E402

Basically need to add the `import bar` back to the test above and add `import bar` to the `if` test

Comment (EricCousineau-TRI) [reply testsuite/E40.py]: Er, I think I fixed it?

Comment (asottile) [reply testsuite/E40.py]: awesome, looks great

File: pycodestyle.py [hunk 2]
```diff
@@ -1067,7 +1067,8 @@ def is_string_literal(line):
             line = line[1:]
         return line and (line[0] == '"' or line[0] == "'")
 
-    allowed_try_keywords = ('try', 'except', 'else', 'finally')
+    allowed_keywords = (
+        'try', 'except', 'else', 'finally', 'with', 'if')
```
Comment (GPHemsley) [inline pycodestyle.py]: Should this also include `elif`?

Comment (EricCousineau-TRI) [reply pycodestyle.py]: Done.

**Pull request merged**